### PR TITLE
537 remove workflows

### LINKED ISSUE #530: Make instances always accessible
We need to make our bim2sim elements which are mostly referred as `instances` in the code available at every point for the webtool.

### LINKED ISSUE #536: cleanup project folder
image

Following folders are not used anymore or it's not clear for what they should be used:
* [x] resources is not clear (currently empty)
* [x] task is not clear (currently empty)

For assets (ifc_example_files , finder and enrichment) I'm currently not sure if we really need them in project directory. Needs further investigation.

### LINKED ISSUE #537: Refactor/remove workflows and make ready for multiple IFCs
This will involve some major changes to address the following problems:
### Mehrere Probleme, die Umbauten erforderlich machen im Tool und gemeinsam gelöst werden sollten:
* [ ] Usage of multiple IFCs (e.g. 1x for BPS, 1x for HVAC) #35 
* [ ] Workflows are not used for what they were created. This leads to confusions. Workflows will be removed, settings and relevant_elements will be moved to plugins and ifc_units will be moved to Playground
* [ ] instances must always be existing reachable for webtool, so they will be moved to playground as well #530
  * [ ] playground must be reachable from every task

* [ ] Config and workflow_settings aren't clean. We will refactor this to the following state:
  * project creation and FolderStrucutre does not automaticly creates a base config  file. Instead a file is loaded if it exists. If not, the file will only be created after all settings have been set through the code. The base for the created config file will be the defined settings + default values of settings. Might be correlated to #511, but #511 seems more to be a user problem, but if not this should solve this

### PR DESCRIPTION
**Pull request by DaJansenGit**
* closes #537 
* closes #536 
* closes #530 

There is one ToDo left for you @veronikarichter [here](https://github.com/BIM2SIM/bim2sim/blob/537_remove_workflows/bim2sim/plugins/PluginEnergyPlus/bim2sim_energyplus/task/ep_idf_postprocessing.py#L114-L117)
Maybe this ToDo are the reasons why the regression tests for EP still fail? Otherwise, we can have a look together.

I might also add some tests, but you can already start your review.

### CODE REVIEW

**Comment by veronikarichter** [comment]
> There is one ToDo left for you @veronikarichter [here](https://github.com/BIM2SIM/bim2sim/blob/537_remove_workflows/bim2sim/plugins/PluginEnergyPlus/bim2sim_energyplus/task/ep_idf_postprocessing.py#L114-L117) Maybe this ToDo are the reasons why the regression tests for EP still fail? Otherwise, we can have a look together.

I fixed the todo and resolved the issues within the regression tests. The pipeline currently fails due to an issue within "pages".

File: bim2sim/task/bps/sb_creation.py [hunk 1]
```diff
@@ -14,33 +14,38 @@
...
+                self.playground.sim_settings.create_external_elements,
+                ifc_cls.ifc_units)
+            bound_instances = self.get_parents_and_children(
+                self.playground.sim_settings, instance_lst, instances)
+            instance_lst = list(bound_instances.values())
+            logger.info(f"Created {len(bound_instances)} bim2sim SpaceBoundary "
+                        f"instances based on IFC file: {ifc_cls.ifc_file_name}")
+            space_boundaries.update({inst.guid: inst for inst in instance_lst})
+        logger.info(f"Created {len(space_boundaries)} bim2sim SpaceBoundary "
+                    f"instances in total for all IFC files.")
```
Comment (veronikarichter) [inline bim2sim/task/bps/sb_creation.py:45]: I would rather not name it ifc_cls as it sounds like these are individual classes within the ifc rather then different ifc_files.
```suggestion
        space_boundaries = {}
        for ifc_file in ifc_files:
            entity_type_dict, unknown_entities = type_filter.run(ifc_file.file)
            instance_lst = self.instantiate_space_boundaries(
                entity_type_dict, instances, ifc_file.finder,
                self.playground.sim_settings.create_external_elements,
                ifc_file.ifc_units)
            bound_instances = self.get_parents_and_children(
                self.playground.sim_settings, instance_lst, instances)
            instance_lst = list(bound_instances.values())
            logger.info(f"Created {len(bound_instances)} bim2sim SpaceBoundary "
                        f"instances based on IFC file: {ifc_file.ifc_file_name}")
            space_boundaries.update({inst.guid: inst for inst in instance_lst})
        logger.info(f"Created {len(space_boundaries)} bim2sim SpaceBoundary "
                    f"instances in total for all IFC files.")
```

File: docs/source/concepts/sim_settings.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
+(simulation_setting)=
+# Simulation settings
+## What are simulation settings?
+`bim2sim` uses prescribed settings which define how a specific simulations 
+should be handled. For basic setup we have a [BaseSimSettings](BaseSimSettings),
+which is extended for the different simulation types (e.g.
+[BuildingSimSettings](BuildingSimSettings)) and this domain simulation type 
+specific settings can also be further specified for each [Plugin](Plugin)
+(e.g. [TEASERSimSettings](TEASERSimSettings) or
+[EnergyPlusSimSettings](EnergyPlusSimSettings)) via clas inheritance.
```
Comment (veronikarichter) [inline docs/source/concepts/sim_settings.md:10]:
```suggestion
[EnergyPlusSimSettings](EnergyPlusSimSettings)) via class inheritance.
```

File: docs/source/concepts/sim_settings.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
+(simulation_setting)=
+# Simulation settings
+## What are simulation settings?
+`bim2sim` uses prescribed settings which define how a specific simulations
```
Comment (veronikarichter) [inline docs/source/concepts/sim_settings.md:4]:
```suggestion
`bim2sim` uses prescribed settings which define how a specific simulation
```

File: docs/source/concepts/sim_settings.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+## What are simulation settings?
+`bim2sim` uses prescribed settings which define how a specific simulations 
+should be handled. For basic setup we have a [BaseSimSettings](BaseSimSettings),
+which is extended for the different simulation types (e.g.
+[BuildingSimSettings](BuildingSimSettings)) and this domain simulation type 
+specific settings can also be further specified for each [Plugin](Plugin)
+(e.g. [TEASERSimSettings](TEASERSimSettings) or
+[EnergyPlusSimSettings](EnergyPlusSimSettings)) via clas inheritance.
+ 
+Each [Plugin](Plugin) has a setting predefined. But the different settings can
```
Comment (veronikarichter) [inline docs/source/concepts/sim_settings.md:12]:
```suggestion
Each [Plugin](Plugin) has a set of predefined settings. But the different settings can 
```

File: docs/source/concepts/sim_settings.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+If you want to overwrite the specified `relevant_elements` you can 
+do this in the same way
+```python
+project.sim_settings.relevant_elements = {<classes of elements of interest>}
+```
+The first way is useful if you want to work in an interactive way, the second is 
+more fitting for automated tasks.
+
+## Want to set up your own workflow?
+The pre implemented workflows hold the needed settings for the simulation with
```
Comment (veronikarichter) [inline docs/source/concepts/sim_settings.md:49]:
```suggestion
The pre-implemented workflows hold the needed settings for the simulation with 
```

File: docs/source/concepts/sim_settings.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+do this in the same way
+```python
+project.sim_settings.relevant_elements = {<classes of elements of interest>}
+```
+The first way is useful if you want to work in an interactive way, the second is 
+more fitting for automated tasks.
+
+## Want to set up your own workflow?
+The pre implemented workflows hold the needed settings for the simulation with 
+the pre implemented Plugins. For further customization you can add your own
```
Comment (veronikarichter) [inline docs/source/concepts/sim_settings.md:50]:
```suggestion
the pre-implemented Plugins. For further customization you can add your own 
```

File: docs/source/overview.md [hunk 7]
```diff
@@ -55,7 +55,7 @@ A project is the main object in `bim2sim` and brings workflow and plugin
 together and allows to run the process of simulation model creation.
 
 **Inputs:**
-* A [workflow](concepts/workflow.md) holds the relevant settings for each type of 
+* A [workflow](concepts/sim_settings.md) holds the relevant settings for each type of
```
Comment (veronikarichter) [inline docs/source/overview.md:58]:
```suggestion
* [SimSettings](concepts/sim_settings.md) hold the relevant settings for each type of 
```

File: test/unit/kernel/task/test_expansiontank.py [hunk 8]
```diff
@@ -59,7 +60,7 @@ def test_expansion_tank_circuit_decision(self):
             expansiontanks.ExpansionTanks.identify_expansion_tanks(graph)
         self.assertEqual(
             1, len(pot_tanks),
-            f"There is 1 expansion tank but ony {len(pot_tanks)} was identified."
+            f"There is 1 expansion tank but {len(pot_tanks)} was identified."
```
Comment (veronikarichter) [inline test/unit/kernel/task/test_expansiontank.py:63]:
```suggestion
            f"There is 1 expansion tank but {len(pot_tanks)} were identified."
```

**Review by veronikarichter — CHANGES_REQUESTED** [review]
Good job! Just some minor fixes required, mainly on typos and documentation.

**Review by veronikarichter — COMMENTED** [review]
Looks fine, bugs are fixed. Just resolve the comments from my previous review and we are ready to merge.

File: bim2sim/kernel/condition.py [hunk 9]
```diff
@@ -50,16 +50,19 @@ def __init__(self, key: str, value_min, value_max, incl_edges: bool = False,
...
         for v in value:
             if self.incl_edges:
                 check_list.append(False if not v or v <= self.valueMin
                                   or v >= self.valueMax else True)
             else:
-                check_list.append(False if not v or v < self.valueMin
+                try:
+                    check_list.append(False if not v or v < self.valueMin
                                   or v > self.valueMax else True)
+                except:
```
[the remark this answers is not in the document]
Comment (DaJansenGit) [reply bim2sim/kernel/condition.py]: this was just for debugging purpose. Solved with af984f27584c12f58c9ab966fb23674b1afa1d7b

File: bim2sim/simulation_settings.py [hunk 10]
```diff
@@ -33,70 +37,59 @@ class AutoSettingNameMeta(type):
...
+    The manager is needed to maintain the different attributes of a simulation
     (e.g. choices) while making the read and write access to the setting still
-    easy. This way you can call workflow.<setting_name> and get the value
-    directly while under workflow.manager.<setting_name> you can still find all
+    easy. This way you can call sim_settings.<setting_name> and get the value
+    directly while under sim_settings.manager.<setting_name> you can still find all
     information.
     """
 
-    def __init__(self, bound_workflow):
```
[the remark this answers is not in the document]
Comment (DaJansenGit) [reply bim2sim/simulation_settings.py:76]: This is needed, to have a bidirectional access and know which simulation settings (e.g. `BuildingSimSettings`) the manager is bound to. 

added docs with bf53b7912eec0a942665f89c191d5150ce2dfa21

File: docs/source/concepts/tasks.md [hunk 11]
```diff
@@ -23,13 +23,23 @@ In default mode all `default_tasks` of the selected Plugin are executed in
...
+As mentioned Tasks can have `reads` and `touches`, which define the input and 
+output of every task. Two of those variables are:
+
+* `instances`
+* `graph`
+
+`instances` holds a dictionary with all [elements](elements) that are currently
+existing at runtime. `graph` holds the [HvacGraph](HvacGraph) and is therefore
+only existing for the HVAC [plugins](plugins). These two variables are stored 
+and hold up2date inside the [Playground](Playground) instance.
```
[the remark this answers is not in the document]
Comment (DaJansenGit) [reply docs/source/concepts/tasks.md:42]: adjust docs

File: bim2sim/task/bps/sb_creation.py (hunk 1 above)
Comment (DaJansenGit) [reply bim2sim/task/bps/sb_creation.py]: solved with 12e7c1e157452167063606fffc3874e9d2b75c68 as this was also done in `common_tasks.py`

**Review by veronikarichter — APPROVED** [review]

**Pull request merged**